src/transactions/test/ClawbackTests.cpp

```diff
@@ -0,0 +1,184 @@
+                REQUIRE(a1.getTrustlineBalance(idr) == 25);
+            }
+            SECTION("allow trust can't set clawback")
+            {
+                REQUIRE_THROWS_AS(gateway.allowTrust(
+                                      idr, a1, TRUSTLINE_CLAWBACK_ENABLED_FLAG),
+                                  ex_ALLOW_TRUST_MALFORMED);
+
+                // show that TRUSTLINE_CLAWBACK_ENABLED_FLAG was the reason the
+                // above op failed
+                gateway.allowTrust(idr, a1, 0);
+            }
+        }
+
+        SECTION("errors")
+        {
+            SECTION("set options")
+            {
+                // can't clear revocable if AUTH_CLAWBACK_ENABLED_FLAG is set
+                REQUIRE_THROWS_AS(
+                    gateway.setOptions(clearFlags(AUTH_REVOCABLE_FLAG)),
+                    ex_SET_OPTIONS_AUTH_REVOCABLE_REQUIRED);
+
+                // a1 doesn't have AUTH_REVOCABLE_FLAG set, so can't add
+                // AUTH_CLAWBACK_ENABLED_FLAG
+                REQUIRE_THROWS_AS(
+                    a1.setOptions(setFlags(AUTH_CLAWBACK_ENABLED_FLAG)),
+                    ex_SET_OPTIONS_AUTH_REVOCABLE_REQUIRED);
+
+                // both can be removed together
+                gateway.setOptions(clearFlags(AUTH_CLAWBACK_ENABLED_FLAG |
+                                              AUTH_REVOCABLE_FLAG));
+            }
+            SECTION("check validity")
+            {
+                // clawback from self
+                REQUIRE_THROWS_AS(gateway.clawback(gateway, idr, 75),
+                                  ex_CLAWBACK_MALFORMED);
+
+                // invalid amount
+                REQUIRE_THROWS_AS(gateway.clawback(a1, idr, 0),
+                                  ex_CLAWBACK_MALFORMED);
+                REQUIRE_THROWS_AS(gateway.clawback(a1, idr, -1),
+                                  ex_CLAWBACK_MALFORMED);
+
+                // invalid asset
+                auto invalidAssets = testutil::getInvalidAssets(gateway);
+                for (auto const& asset : invalidAssets)
+                {
+                    REQUIRE_THROWS_AS(gateway.clawback(a1, asset, 75),
+                                      ex_CLAWBACK_MALFORMED);
+                }
+            }
+            SECTION("apply")
+            {
+                SECTION("no trust")
+                {
+                    REQUIRE_THROWS_AS(gateway.clawback(root, idr, 75),
+                                      ex_CLAWBACK_NO_TRUST);
+                }
+                SECTION("not clawback enabled")
+                {
+                    gateway.setOptions(clearFlags(AUTH_CLAWBACK_ENABLED_FLAG));
+                    root.changeTrust(idr, 100);
+                    gateway.pay(root, idr, 100);
+                    REQUIRE_THROWS_AS(gateway.clawback(root, idr, 75),
+                                      ex_CLAWBACK_NOT_CLAWBACK_ENABLED);
+
+                    REQUIRE(a1.getTrustlineBalance(idr) == 100);
```

Comment:
This test must have an error in it. You're operating on `root` but checking on `a1`. I assume you wanted to operate on `a1` judging by the rest of this file.

Reply:
Fixed. I am operating on `root` since it has a trustline without clawback enabled.